### PR TITLE
Support any request method for remote_resource.as_json

### PR DESCRIPTION
**Pull request by kamaradclimber**
- support post and other request methods
- accept any 2xx result (instead of just 200). This is a semi-breaking
  change.
- accept body passed for non GET requests

Change-Id: I8cf6b61421ba5290f928478a808806c4bac53adb

### CODE REVIEW

File: lib/consul/async/json_endpoint.rb [hunk 1]
```diff
@@ -167,9 +172,10 @@ def fetch
           conn: EventMachine::HttpRequest.new(conf.url, options)
         }
         cb = proc do
-          http = connection[:conn].get(build_request)
+          request_method = conf.request_method.to_sym
+          http = connection[:conn].send(request_method, build_request)
           http.callback do
-            if enforce_json_200 && http.response_header.status != 200 && http.response_header['Content-Type'] != 'application/json'
+            if enforce_json_200 && !(200..299).include?(http.response_header.status) && http.response_header['Content-Type'] != 'application/json'
```
Comment (Aestek) [inline lib/consul/async/json_endpoint.rb]: It seems like you're generating an array with a hundred entries and then testing each individual ones. This is called on each response, and there can be a lot of them when watching _everything_ why not just `http.response_header.status => 200 && http.response_header.status < 300` ?

Comment (Aestek) [reply lib/consul/async/json_endpoint.rb]: Digging ruby magic it seems this can just be `http.response_header.status.between?(200, 299)`

Comment (kamaradclimber) [reply lib/consul/async/json_endpoint.rb]: `(200..299)` is a Range which has a specific implementation of include?
(https://github.com/ruby/ruby/blob/132b7eb104ed2f5d9966adb2f0edacb2406f40fb/range.c#L1425)

**Comment by kamaradclimber** [comment]
Use case I have is the mesos api where all actions (read or write) are sent through POSTs to /api/v1 with a body giving the type of operation.
This api returns 201 on every successful request.

**Pull request merged**